packages/core/echo/echo-schema/src/handler/ref.ts

```diff
@@ -14,7 +14,28 @@ import { type Ref } from '../types';
 /**
  * The `$id` field for an ECHO reference schema.
  */
-export const JSON_SCHEMA_ECHO_REF_ID = '/schemas/echo/ref';
+const JSON_SCHEMA_ECHO_REF_ID = '/schemas/echo/ref';
+
+const DXN = /dxn:type:(.+)/;
+
+export const getSchemaReference = (property: any): string | undefined => {
+  const { $id, reference: { schema: { $ref } = {} } = {} } = property;
+  if ($id === JSON_SCHEMA_ECHO_REF_ID) {
+    const match = $ref?.match(DXN);
+    return match?.[1];
+  }
+};
+
+export const setSchemaReference = (property: any, schema: string) => {
+  Object.assign(property, {
```

Comment:
This is not right, because this will leave any other fields on `property` intact.

e.g. 
```ts
const property = { type: 'number' };
Object.assign(property, { $id, reference })
property === { type: 'number', $id, reference }
```

i would instead re-assign the whole object: `schema.properties[key] = { $id, reference }`